pkg/test/step.go

```diff
@@ -273,21 +278,27 @@ func (s *Step) CheckResource(expected runtime.Object, namespace string) []error
 		actual := unstructured.Unstructured{}
 		actual.SetGroupVersionKind(gvk)
 
-		err = cl.Get(context.TODO(), client.ObjectKey{
+		if err := cl.Get(context.TODO(), client.ObjectKey{
 			Namespace: namespace,
 			Name:      name,
-		}, &actual)
+		}, &actual); err != nil {
+			return append(testErrors, err)
+		}
 
 		actuals = append(actuals, actual)
 	} else {
-		actuals, err = list(cl, gvk, namespace)
+		m, err := meta.Accessor(expected)
```

Comment:
I'm a little afraid that `Accessor()` might fail in case `expected` is not of a type [expected by it](https://github.com/kubernetes/apimachinery/blob/v0.28.2/pkg/api/meta/meta.go#L96) (`metav1.Object` or `metav1.ObjectMetaAccessor`) but was previously good enough for `CheckResource` (`runtime.Object`). The different apimachinery types always make my head spin. Did you perhaps look in detail at uses of this method and could confirm with some confidence that we'll be OK?

Reply:
I think thats a valid concern. I would imagine that it could cause an issue in the case its not of type metav1.Object or metav1.ObjectMetaAccessor. Wouldn't this issue already exist though since `Namespaced()` is being called within `CheckResources()`, which is then calling `Accessor()` with `expected` as its argument? 

ttps://github.com/kudobuilder/kuttl/blob/main/pkg/test/utils/kubernetes.go#L338-L343

Reply:
Great point, this is exactly the kind of argument I was looking for.

Sounds like it might make sense to change the `CheckResource()` signature to be more strict, then, but this is definitely not in scope of this PR.